### ISSUE TITLE
[Bug Report] Memory leak happend when call run_search multiple times

### ISSUE THREAD

**Issue by sunface** [quickwit-oss/tantivy#186]
I changed the function run_search_cli  to 
```
pub fn run_search_cli(matches: &ArgMatches) -> Result<(), String> {
    let index_directory = PathBuf::from(matches.value_of("index").unwrap());
    let query = matches.value_of("query").unwrap();
    loop {
        run_search(&index_directory, &query).map_err(|e| format!("{:?}", e));
        let ten_secs = time_std::Duration::from_secs(10);
        thread::sleep(ten_secs);
    }
    Ok(())
    // run_search(&index_directory, &query).map_err(|e| format!("{:?}", e))
}
```

In MacOS Sierra, from command line, I call `tantivy search -i wikipedia-index -q "barack obama" `.
And then the memory leak happens, the memory goes from 1MB to 36MB in one hour, about 10KB/second leak rate

**Comment by fulmicoton** [comment]
Thanks. Are you on master ?

**Comment by fulmicoton** [comment]
Also can you confirm that this is anonymous memory and not page cache?

**Comment by sunface** [comment]
1. On version 0.3.1
2.I'm afrait not, just new to rust and tantivy :D, although i'm trying to study the source code of tantivy,by the way,very nice project and source code!

**Comment by sunface** [comment]
Memory leak happens in both Debug mode and Release Mode

**Comment by sunface** [comment]
If I leave out the loop interval,then memory leak become crazy quick,  in just several seconds, memory goes up to 500MB

**Comment by fulmicoton** [comment]
how do you measure memory going up?

**Comment by sunface** [comment]
from top,and the memory will never goes down

**Comment by sunface** [comment]
Just now,I've tried 0.4.0,It seems ok now

**Comment by sunface** [comment]
Ok,problem resolved in 0.4.0, thats awesome,thanks for your help

**Comment by fulmicoton** [comment]
Oh I see. This must be the bug in crossbeam. Yes it has been "workaround" in 0.4.0.

For the info, even for users using 0.3.1, it is not an "horrible" leak. It only happens if you reopen the index over and over.
